pipelines/skylab/multiome/atac.wdl

```diff
@@ -195,33 +191,128 @@ task BWAPairedEndAlignment {
+    mkdir "output_dir"
+    mkdir "input_dir"
+    
+    echo "Move R1, R3 and reference files to input directory."
+    R1=""
+    echo "R1"
+    for fastq in "${R1_ARRAY[@]}"; do mv "$fastq" input_dir; R1+=`basename $fastq`" "; done
+    echo $R1
+    R3=""
+    echo "R3"
+    for fastq in "${R3_ARRAY[@]}"; do mv "$fastq" input_dir; R3+=`basename $fastq`" "; done
+    echo $R3
+
+    mv $REF_DIR input_dir
+
+    echo "List of files in input directory"
+    ls input_dir
+    
+    # multiome-practice-may15_arcgtf, trimmed_adapters.fastq.gz
+    PREFIX=~{output_base_name}
+    SUFFIX=~{suffix}
+
+    I1=""
+    R2=""
+    
+    echo "REF_PAR_DIR:" $REF_PAR_DIR
+    REF=$REF_PAR_DIR/genome.fa
+    
+    PARAMS='+R "@RG\tID:~{read_group_id}\tSM:~{read_group_sample_name}" +C'
+    
+    INPUT_DIR=$file_path/input_dir
+    OUTPUT_DIR=$file_path/output_dir
+    
+    input_to_config="INPUT_DIR=\"${INPUT_DIR}\"\nOUTPUT_DIR=\"${OUTPUT_DIR}\"\nPREFIX=\"${PREFIX}\"\nSUFFIX=\"${SUFFIX}\"\n"
+    other_to_add="R1=\"${R1}\"\nR2=\"${R2}\"\nR3=\"${R3}\"\nI1=\"${I1}\"\nREF=\"${REF}\"\n"
+    params="PARAMS=\'${PARAMS}\'"
+    
+    printf "%b" "$input_to_config"
+    printf "%b" "$other_to_add"
+    echo $params
+    
+    # cd into fq2sortedbam
+    cd /usr/temp/Open-Omics-Acceleration-Framework/pipelines/fq2sortedbam
+    # remove the first part of config
+    tail -10 config > config
+    # add inputs to config file (this file is needed to run bwa-mem2 in this specific code"
+    printf "%b" "$input_to_config" | tee -a config
+    printf "%b" "$other_to_add" | tee -a config
+    echo $params | tee -a config
+    echo "CONFIG"
+    cat config
+    # run bwa-mem2
+    echo "Run distributed BWA-MEM2"
+    ./run_bwa.sh multifq
+    echo "Done running distributed BWA-MEM2"
+    echo "List of files in output directory"
+    ls $OUTPUT_DIR
+    cd $OUTPUT_DIR
+    
+    # remove all files except for final and text file 
+    echo "Remove all files except for final bam file and log files"
+    ls | grep -xv final.sorted.bam | grep -v .txt$ | xargs rm
+
+    echo "List of files in output directory after removal"
+    ls
+    
+    # rename file to this
+    mv final.sorted.bam ~{bam_aligned_output_name}
+        
+    # save output logs for bwa-mem2
+    mkdir output_logs
+    mv *txt output_logs
+    tar -zcvf /cromwell_root/output_distbwa_log.tar.gz output_logs  
+    
+    # move bam file to /cromwell_root
+    mv ~{bam_aligned_output_name} /cromwell_root
+    ls /cromwell_root
 
-    # align w/ BWA: -t for number of cores
-    bwa-mem2 \
-    mem \
-    -R "@RG\tID:~{read_group_id}\tSM:~{read_group_sample_name}" \
-    -C \
-    -t ~{nthreads} \
-    $REF_DIR/genome.fa \
-    ~{read1_fastq} ~{read3_fastq} \
-    | samtools view -bS - > ~{bam_aligned_output_name}
   >>>
 
   runtime {
     docker: docker_image
-    disks: "local-disk ${disk_size} HDD"
+    disks: "local-disk ${disk_size} SSD"
     cpu: nthreads
+    cpuPlatform: "Intel Ice Lake"
```

Comment:
Make a task input so that Smart-tests work(Smart-tests use older Cromwell version that doesn't have Intel Ice Lake).